File: frontend/src/services/homeworkService.js
import api from './api'

export const homeworkService = {
  create: (payload) => api.post('/homework', payload),
  findByClass: (classId) => api.get(`/homework/class/${classId}`),
  update: (id, payload) => api.put(`/homework/${id}`, payload),
  remove: (id) => api.delete(`/homework/${id}`),
}

File: backend/src/main/java/com/school/service/HomeworkService.java
package com.school.service;

import com.school.dto.request.HomeworkRequest;
import com.school.dto.response.HomeworkResponse;

import java.util.List;

public interface HomeworkService {
    HomeworkResponse create(HomeworkRequest request, String teacherId);
    List<HomeworkResponse> findByClass(String classId);
    HomeworkResponse update(String id, HomeworkRequest request);
    void delete(String id);
}